openapi.yaml

```diff
@@ -394,10 +443,85 @@ paths:
         "500":
           description: Internal server error
 
+  /apiKey/v2/test-targets/{testTargetId}/test-reports:
+    get:
+      summary: Retrieve paginated information about test reports.
+      description: Allow fetching the history of test reports for your test target.
+      parameters:
+        - in: path
+          name: testTargetId
+          required: true
+          schema:
+            type: string
+            format: uuid
+          description: ID of the test target for which to fetch the history for
+        - in: query
+          name: key
+          required: false
+          schema:
+            type: object
+            properties:
+              createdAt:
+                type: string
+                format: date-time
+                description: The timestamp of the key of the next page to fetch - See [Keyset Pagination](https://use-the-index-luke.com/no-offset)
+                example: "2024-09-06T13:01:51.686Z"
+        - in: query
+          name: filter
+          required: false
+          schema:
+            type: object
+            properties:
+              environmentId:
```

Comment:
theoretically this allows OTHER filters as well dynamically based on the type of the testReport -> I think the only one that might make sense for now is the environmentId, so I would only add more here/make it dynamic in the docs if they are requested